packages/bloc/lib/src/cubit.dart

```diff
@@ -148,12 +147,11 @@ abstract class Cubit<State> extends Stream<State> {
   /// handlers.
   @override
   StreamSubscription<State> listen(
-    void Function(State) onData, {
-    Function onError,
-    void Function() onDone,
-    bool cancelOnError,
+    void Function(State)? onData, {
+    Function? onError,
+    void Function()? onDone,
+    bool? cancelOnError,
   }) {
-    _controller ??= StreamController<State>.broadcast();
     return _controller.stream.listen(
       onData,
```

Comment:
I am surprised that this isn't an error, this shouldn't be a nullable argument as far as I can tell - https://api.dart.dev/stable/2.10.0/dart-async/Stream/listen.html.

Reply:
looks like it is https://github.com/dart-lang/sdk/blob/34e8b4229efec853ae171dc1569c03b1d053c58f/sdk/lib/async/stream.dart#L507 😕

Reply:
Ah it looks like that is because we allow you to set this lazily. Weird (and unfortunate I think...), but fair enough :).